File: client/java/src/avt/AvatarService.java
package avt;

import java.io.IOException;
import main.Canvas;
import main.GameMidlet;

public final class AvatarService extends IService {
   private static AvatarService instance;

   public static AvatarService gI() {
      if (instance == null) {
         instance = new AvatarService();
      }

      return instance;
   }

   public final void getBigData() {
      this.createMessage((byte)-11);
      this.writeInt(LoginScr.s);
      this.sendMessage();
   }

   public final void getBigImage(short var1) {
      this.createMessage((byte)-14);
      this.writeShort(var1);
      this.sendMessage();
      Canvas.startWaitDlg(T.enemyFirstFire);
   }

   public final void getImageData() {
      this.createMessage((byte)-15);
      this.sendMessage();
      Canvas.startWaitDlg(T.enemyFirstFire);
   }

   public final void getAvatarPart() {
      this.createMessage((byte)-16);
      this.sendMessage();
      Canvas.startWaitDlg(T.enemyFirstFire);
   }

   public final void getItemInfo() {
      this.createMessage((byte)-37);
      this.sendMessage();
      Canvas.startWaitDlg(T.enemyFirstFire);
   }

   public final void getMapItemType() {
      this.createMessage((byte)-40);
      this.sendMessage();
   }

   public final void getMapItem() {
      this.createMessage((byte)-41);
      this.sendMessage();
   }

   public final void doFeel(int var1) {
      if (GameMidlet.CLIENT_TYPE == 9 || GameMidlet.CLIENT_TYPE == 11) {
         this.createMessage((byte)57);
         this.writeByte(var1);
         this.sendMessage();
      }

   }

   public final void doBuyItem(int var1, int var2) {
      System.out.println("DEBUG BUYITEM: AvatarService.doBuyItem(id=" + var1 + ", moneyType=" + var2 + ") packet -24");
      this.createMessage((byte)-24);

      try {
         super.m.writer().writeShort(var1);
         super.m.writer().writeByte(var2);
      } catch (Exception var4) {
      }

      this.sendMessage();
   }

   public final void doCreateHome(short[] var1, int var2) {
      this.createMessage((byte)-46);

      try {
         super.m.writer().writeShort(var2);
         super.m.writer().writeShort(var1.length);

         for(var2 = 0; var2 < var1.length; ++var2) {
            super.m.writer().writeByte(var1[var2]);
         }

         super.m.writer().writeShort(0);
      } catch (Exception var4) {
      }

      this.sendMessage();
   }

   public final void doBuyItemHouse(MapItem var1) {
      System.out.println("doBuyItemHouse; " + var1.typeID);
      this.createMessage((byte)-74);

      try {
         super.m.writer().writeShort(var1.typeID);
         super.m.writer().writeByte(var1.x / 24);
         super.m.writer().writeByte(var1.y / 24);
         super.m.writer().writeByte(var1.type);
      } catch (IOException var3) {
      }

      this.sendMessage();
   }

   public final void doJoinHouse(int var1) {
      Canvas.startWaitDlg();
      this.createMessage((byte)-65);
      this.writeInt(var1);
      this.sendMessage();
   }

   public final void dodelItem(MapItem var1) {
      this.createMessage((byte)-66);

      try {
         super.m.writer().writeShort(var1.typeID);
         super.m.writer().writeByte(var1.x / 24);
         super.m.writer().writeByte(var1.y / 24);
         super.m.writer().writeByte(var1.dir);
      } catch (Exception var3) {
      }

      this.sendMessage();
   }

   public final void getTypeHouse(int var1) {
      this.createMessage((byte)-67);
      this.writeByte(var1);
      this.sendMessage();
   }

   public final void doKickOutHome(int var1) {
      this.createMessage((byte)-69);
      this.writeInt(var1);
      this.sendMessage();
   }

   public final void doRequestExpicePet(int var1) {
      this.createMessage((byte)-70);
      this.writeInt(var1);
      this.sendMessage();
   }

   public final void doSortItem(int var1, int var2, int var3, int var4, int var5, int var6) {
      this.createMessage((byte)-71);

      try {
         super.m.writer().writeShort(var1);
         super.m.writer().writeByte(var2);
         super.m.writer().writeByte(var3);
         super.m.writer().writeByte(var4);
         super.m.writer().writeByte(var5);
         super.m.writer().writeByte(var6);
      } catch (IOException var8) {
      }

      this.sendMessage();
   }

   public final void doSetPassMyHouse(String var1, int var2, int var3) {
      this.createMessage((byte)-75);

      try {
         super.m.writer().writeByte(var3);
         super.m.writer().writeUTF(var1);
         if (var3 == 1) {
            super.m.writer().writeInt(var2);
         }
      } catch (IOException var5) {
      }

      this.sendMessage();
   }

   public final void doGetImgIcon(short var1) {
      this.createMessage((byte)-80);
      this.writeShort(var1);
      this.sendMessage();
   }

   public final void doRequestEffectData(short var1) {
      this.createMessage((byte)-84);
      this.writeByte(var1);
      this.sendMessage();
   }

   public final void doJoinHouse4(int var1) {
      this.createMessage((byte)-104);
      this.writeInt(var1);
      this.sendMessage();
   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/AvatarService.cs
using System;
using System.IO;

public class AvatarService
{
	protected static AvatarService instance;

	public static AvatarService gI()
	{
		if (instance == null)
		{
			instance = new AvatarService();
		}
		return instance;
	}

	public void send(Message m)
	{
		Session_ME.gI().sendMessage(m);
		m.cleanup();
	}

	public void getBigData()
	{
		Message m = new Message((sbyte)(-11));
		send(m);
	}

	public void getBigImage(short id)
	{
		Message message = new Message((sbyte)(-14));
		try
		{
			message.writer().writeShort(id);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
		Canvas.startWaitDlg(T.getData);
		MsgDlg.isBack = false;
	}

	public void getImageData()
	{
		Message m = new Message((sbyte)(-15));
		send(m);
		MsgDlg.isBack = false;
	}

	public void getAvatarPart()
	{
		Message m = new Message((sbyte)(-16));
		send(m);
		MsgDlg.isBack = false;
	}

	public void getItemInfo()
	{
		Message m = new Message((sbyte)(-37));
		send(m);
		MsgDlg.isBack = false;
	}

	public void getMapItemType()
	{
		Message m = new Message((sbyte)(-40));
		send(m);
	}

	public void getMapItem()
	{
		Message m = new Message((sbyte)(-41));
		send(m);
	}

	public void doFeel(int focusFeel)
	{
		if (GameMidlet.CLIENT_TYPE != 9 && GameMidlet.CLIENT_TYPE != 11)
		{
			return;
		}
		Message message = new Message((sbyte)57);
		try
		{
			message.writer().writeByte(focusFeel);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doBuyItem(int part, int typeBuy)
	{
		Message message = new Message((sbyte)(-24));
		try
		{
			message.writer().writeShort((short)part);
			message.writer().writeByte((sbyte)typeBuy);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doGetTileInfo()
	{
		Message m = new Message((sbyte)(-43));
		send(m);
	}

	public void doCreateHome(short[] map, int type)
	{
		Message message = new Message((sbyte)(-46));
		try
		{
			message.writer().writeShort((short)type);
			message.writer().writeShort((short)map.Length);
			for (int i = 0; i < map.Length; i++)
			{
				message.writer().writeByte((sbyte)map[i]);
			}
			message.writer().writeShort(0);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doBuyItemHouse(MapItem map)
	{
		Message message = new Message((sbyte)(-74));
		try
		{
			message.writer().writeShort(map.typeID);
			message.writer().writeByte((sbyte)(map.x / 24));
			message.writer().writeByte((sbyte)(map.y / 24));
			message.writer().writeByte((sbyte)map.type);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doJoinHouse(int iddb)
	{
		Canvas.startWaitDlg();
		Message message = new Message((sbyte)(-65));
		try
		{
			message.writer().writeInt(iddb);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void dodelItem(MapItem pos)
	{
		Message message = new Message((sbyte)(-66));
		try
		{
			message.writer().writeShort(pos.typeID);
			message.writer().writeByte((sbyte)(pos.x / 24));
			message.writer().writeByte((sbyte)(pos.y / 24));
			message.writer().writeByte(pos.dir);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void getTypeHouse(int type)
	{
		Message message = new Message((sbyte)(-67));
		try
		{
			message.writer().writeByte(type);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doKickOutHome(int iddb)
	{
		Message message = new Message((sbyte)(-69));
		try
		{
			message.writer().writeInt(iddb);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doRequestExpicePet(int idUser)
	{
		Message message = new Message((sbyte)(-70));
		try
		{
			message.writer().writeInt(idUser);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doSortItem(int anchor, int x, int y, int x2, int y2, int dir)
	{
		Message message = new Message((sbyte)(-71));
		try
		{
			message.writer().writeShort((short)anchor);
			message.writer().writeByte((sbyte)x);
			message.writer().writeByte((sbyte)y);
			message.writer().writeByte((sbyte)x2);
			message.writer().writeByte((sbyte)y2);
			message.writer().writeByte((sbyte)dir);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doGetTileMap()
	{
		Message m = new Message((sbyte)(-73));
		send(m);
	}

	public void doSetPassMyHouse(string text, int idUser, int type)
	{
		Message message = new Message((sbyte)(-75));
		try
		{
			message.writer().writeByte((sbyte)type);
			message.writer().writeUTF(text);
			if (type == 1)
			{
				message.writer().writeInt(idUser);
			}
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doSMSServerLoad(string link)
	{
		Message message = new Message((sbyte)(-76));
		try
		{
			message.writer().writeUTF(link);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doGetImgIcon(short id)
	{
		Message message = new Message((sbyte)(-80));
		try
		{
			message.writer().writeShort(id);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doRequestEffectData(short id3)
	{
		Message message = new Message((sbyte)(-84));
		try
		{
			message.writer().writeByte(id3);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doJoinHouse4(int id)
	{
		Message message = new Message((sbyte)(-104));
		try
		{
			message.writer().writeInt(id);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}
}
